### ISSUE TITLE
trino-python-client returns dtype string for datetime/timestamp fields

### ISSUE THREAD

**Issue by RJ3** [trinodb/trino-python-client#32]
presto-client returns data type of string for a timestamp field.
Setup:  
 * PrestoSQL 333  
 * PostgreSQL 12.1  
 * presto-client 0.300.0

Tested:  
1. Using the PrestoSQL JDBC driver, the field is returned as a Java timestamp.
2. Using `psycopg2` driver with SQLalchemy directly to Postgres, the field is returned as a python datetime object or pandas timestamp object.  
3. **Using presto-client 0.300, the field is returned as a string.**

Expectation:   
The field should be returned as a python datetime object or pandas timestamp object, akin to the PostgreSQL driver.

Note:   
The issue has also been reported for pyhive (no longer under current development). I have not tested PrestoDB's version to see if the other team has also worked on the topic.

**Comment by findepi** [comment]
The biggest question is how to avoid breaking users. 
For now we should implement napping to datetime behing a toggle (init param).
At some point, we probably should consdier switching the default, as we do not want to maintain inferior representation for ever.
cc @electrum

**Comment by alexbrisan** [comment]
@findepi I would like to contribute a fix for this. Would you like me to detail a fix first, or just submit a Pull Request?

**Comment by findepi** [comment]
@alexbrisan awesome!

it's worthwhile to have a conversation before doing work, if the work would take a lot of time.
Perhaps this won't be the case here, will it?

Things to consider

-  what do other drivers do? eg. postgresql's or oracle's?
- do we have a toggle to switch between old & new behavior?
- i am not expert on date/time/timezones handling in Python, but it seems the standard library doesn't cover all the use-cases (hence `pytz` exists). 
   - this is why checking with oracle's driver can be interesting. PostgreSQL doesn't have SQL standard-conforming `timestamp with time zone` data type and both Trino and Oracle do.

**Comment by alexbrisan** [comment]
I will take a look at the oracle one and come up with a plan!

**Comment by john-bodley** [comment]
Thanks for authoring the PR.

I was running into an issue with Apache Superset related to inconsistencies in how the frontend handled timestamps. We leverage Pandas for reading a SQL query into a dataframe and serializing it to JSON.

For context here's difference between MySQL and Trino, 

```python
import pandas as pd 
from sqlalchemy import create_engine

pd.read_sql_query(
    sql="SELECT TIMESTAMP('2022-01-01 00:00:00') AS ts", 
    con=create_engine("mysql://root@localhost:3360"),
).to_json()

pd.read_sql_query(
    sql="SELECT TIMESTAMP '2022-01-01 00:00:00' AS ts", 
    con=create_engine("trino://localhost: 8080"),
).to_json()
```

which outputs `{"ts":{"0":1640995200000}}` (which infers UTC per the EPOCH definition) and `{"ts":{"0":"2022-01-01 00:00:00.000"}}` (which is in the local timezone) respectively and thus are treated differently in JavaScript:

```javascript
new Date(1640995200000)
> Sat Jan 01 2022 13:00:00 GMT+1300 (New Zealand Daylight Time)

new Date("2022-01-01 00:00:00.000")
> Sat Jan 01 2022 00:00:00 GMT+1300 (New Zealand Daylight Time)
```

**Comment by mdesmet** [comment]
I think this issue can be closed as we merged #160.

**Comment by ebyhr** [comment]
@mdesmet Closed. Thanks for your contribution!
